src/features/class-parsely-support.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Parsely_Support class file
+ *
+ * @package wp-curate
+ */
+
+namespace Alley\WP\WP_Curate\Features;
+
+use Alley\WP\Types\Feature;
+use Parsely\RemoteAPI\Analytics_Posts_API;
+
+/**
+ * Add support for Parsely, if the plugin is installed.
+ */
+final class Parsely_Support implements Feature {
+	/**
+	 * Set up.
+	 */
+	public function __construct() {}
+
+	/**
+	 * Boot the feature.
+	 */
+	public function boot(): void {
+		if ( ! class_exists( 'Parsely\Parsely' ) ) {
+			return;
+		}
+		// Elsewhere in the plugin, we'll use $GLOBALS['parsely'], but it is not available here.
+		$parsely = new \Parsely\Parsely();
+		// If we don't have the API secret, we can't use the Parsely API.
+		if ( ! $parsely->api_secret_is_set() ) {
+			return;
+		}
+		add_filter( 'wp_curate_use_parsely', '__return_true' );
+		add_filter( 'wp_curate_trending_posts_query', [ $this, 'add_parsely_trending_posts_query' ], 10, 2 );
+	}
+
+	/**
+	 * Gets the trending posts from Parsely.
+	 *
+	 * @param array<number>        $posts The posts, which should be an empty array.
+	 * @param array<string, mixed> $args The WP_Query args.
+	 * @return array<number> Array of post IDs.
+	 */
+	public function add_parsely_trending_posts_query( array $posts, array $args ): array {
+		$parsely = $GLOBALS['parsely'];
+		if ( ! $parsely->api_secret_is_set() ) {
+			return $posts;
+		}
+		$trending_posts = $this->get_trending_posts( $args );
+		return $trending_posts;
+	}
+
+	/**
+	 * Gets the trending posts from Parsely.
+	 *
+	 * @param array<string, mixed> $args The WP_Query args.
+	 * @return array<int> An array of post IDs.
+	 */
+	public function get_trending_posts( array $args ): array {
+		$parsely_options = $GLOBALS['parsely']->get_options();
+		/**
+		 * Filter the period start for the Parsely API.
+		 *
+		 * @param string $period_start The period start.
+		 */
+		$period_start = apply_filters( 'wp_curate_parsely_period_start', '1d' );
```

Comment:
Could pass `$args` here too, so implementers could make different decisions about the period based on the arguments passed to the function.